// Copyright 2024 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto2";

message ArbitrarySyscall {
  optional uint32 number = 1;
  optional bytes arguments = 2;
}

enum PortNumberMach {
  CASE_0 = 0;
  CASE_1 = 1;
  CASE_2 = 2;
  CASE_3 = 3;
  CASE_4 = 4;
  CASE_5 = 5;
  CASE_6 = 6;
  CASE_7 = 7;
  CASE_8 = 8;
  CASE_9 = 9;
  CASE_10 = 10;
  CASE_11 = 11;
  CASE_12 = 12;
  CASE_13 = 13;
  CASE_14 = 14;
  CASE_15 = 15;
  CASE_16 = 16;
  CASE_17 = 17;
  CASE_18 = 18;
  CASE_19 = 19;
  CASE_20 = 20;
  CASE_NEGATIVE = -1;
}
